test/list/ListProxyTest.js

```diff
@@ -269,16 +265,22 @@ describe('ListProxyTest', function () {
     });
 
     it('remove entry listener', async function () {
-        this.timeout(5000);
+        const itemRemovedTriggered = deferredPromise();
         const registrationId = await listInstance.addItemListener({
             itemRemoved: function (itemEvent) {
-                expect(itemEvent.name).to.be.equal('test');
-                expect(itemEvent.item).to.be.equal(1);
-                expect(itemEvent.eventType).to.be.equal(ItemEventType.REMOVED);
-                expect(itemEvent.member).to.not.be.equal(null);
+                try {
+                    expect(itemEvent.name).to.be.equal('test');
+                    expect(itemEvent.item).to.be.equal(1);
+                    expect(itemEvent.eventType).to.be.equal(ItemEventType.REMOVED);
+                    expect(itemEvent.member).to.not.be.equal(null);
+                    itemRemovedTriggered.resolve();
+                } catch (err) {
+                    itemRemovedTriggered.reject(err);
+                }
             }
         });
         const removed = await listInstance.removeItemListener(registrationId);
         expect(removed).to.be.true;
+        await itemRemovedTriggered.promise;
```

Comment:
We just register/deregister listener here, there should not be any waiting

Reply:
I've just noticed the problem and fixed the test. Could you take another look?